src/fsharp/fsi/fsi.fs

```diff
@@ -1859,20 +1861,57 @@ type internal FsiInteractionProcessor
             stopProcessingRecovery e range0    
             None
 
+    let tempDir = @"D:\temp\fsi"
+    let paketPrefix = "paket: "
+    let paketExePath = @"D:\temp\fsi\.paket\paket.exe"
+
+    let resolvePaket ctok istate errorLogger m =
+        if not needsPaketInstall then istate else
+
+        let startInfo = ProcessStartInfo()
+        startInfo.FileName <- paketExePath
+        startInfo.WorkingDirectory <- tempDir
+        startInfo.Arguments <- "install --generate-load-scripts"
+        startInfo.UseShellExecute <- false
+        let p = Process.Start(startInfo)
+        p.WaitForExit()
+        if p.ExitCode <> 0 then
+            failwithf "Paket restore failed."
```

Comment:
Yes. Though it's just a hack right now :)  I'm going to iterate on it to make it activate at design-time